inventory/any.cf

```diff
@@ -772,20 +814,57 @@ bundle agent cfe_autorun_inventory_dmidecode
       # processor-family
       # processor-frequency
       # processor-manufacturer
-      # system-uuid
       #"processor-version": "CPU model" <- Collected by default, but not by iterating over the list
 
       "dmivars" slist => getindices(dmidefs);
 
     have_dmidecode::
       "decoder" string => "$(inventory_control.dmidecoder)";
 
+    have_dmidecode._stdlib_path_exists_awk.!(redhat_4|redhat_3)::
+      # Awk script from https://kb.vmware.com/s/article/53609
+      # Edited only to add "-t1" (an improvement tested on RHEL 4/5/6/7 and FreeBSD)
+      # and to take out the "UUID: " prefix in the output.
+      # This works on a superset of systems where dmidecode -s system-uuid works,
+      # e.g. RHEL 5 with dmidecode-2.7-1.28.2.el5 where system-uuid is not one of the valid keywords;
+      # also, this returns the correct UUID on systems (such as VMWare VMs with hardware version 13)
+      # where dmidecode -s system-uuid shows the wrong UUID.  Some such VMWare VMs also show the
+      # wrong UUID in sysfs, which is why we prefer the "dmidecode | awk" version to sysfs for UUID.
+      # (We still need to check sysfs for UUID to handle hosts without dmidecode such as CoreOS.)
+      "dmi[system-uuid]"
```

Comment:
I know the comment above this is already way too long, but one important piece is missing there -- and example of the output from `dmidecode -u -t1`. That would make the *AWK* script much easier to understand and review.

Reply:
It won't make it as much easier as you think, and it's long enough that I don't think it belongs inline in that file, but here is an example.  It's easy to produce on any box with dmidecode anyway.

```
# dmidecode 3.0
Getting SMBIOS data from sysfs.
SMBIOS 2.5 present.

Handle 0x0001, DMI type 1, 27 bytes
	Header and Data:
		01 1B 01 00 01 02 03 04 88 20 6F B5 62 27 4B 7A
		83 1D 5C 58 59 B3 25 46 06 00 05
	Strings:
		69 6E 6E 6F 74 65 6B 20 47 6D 62 48 00
		"innotek GmbH"
		56 69 72 74 75 61 6C 42 6F 78 00
		"VirtualBox"
		31 2E 32 00
		"1.2"
		30 00
		"0"
		56 69 72 74 75 61 6C 20 4D 61 63 68 69 6E 65 00
		"Virtual Machine"

```

And the output of the script on the same box:

```
88206FB5-6227-4B7A-831D-5C5859B32546
```

The code is on the linked page, though.  I think that's the proper place where the documentation should be improved.

I've submitted feedback on the linked page saying:

> Using dmidecode -u -t1 makes for a lot less output for awk to parse.  Also, it makes the output size manageable enough that you could include an example output of dmidecode -u -t1, which would make the script much easier to understand.

Reply:
Thanks for the example. It will at least be available here for the record. You're right it doesn't belong in that file.